### DISCUSSION TITLE
Embedding nats server in other languages than go

### DISCUSSION THREAD

**Discussion by isaaca26** [nats-io/nats-server#5762] [Q&A]
Hey Everyone

I'm wondering if anyone has successfully been able to embed a nats server in languages other than go? One idea I had was to potentially compile the go code to a c-shared library using the -buildmode=c-shared build argument. This would then allow other languages to potentially embed the nats server in languages other than go. I'm not sure if this is even possible without a re-architecture of the whole nats server code base or maybe some straight forward changes. If anyone has any other ideas, I would really appreciate it.

**Comment by niklasfp** [comment]
I made a poc on embedding nats in .net, https://github.com/niklasfp/dotnet-nats-server, it utilises a c-shared lib compiled version of nats server. 

>[!NOTE]
> It's a poc, and I'm in no way an experienced go developer 😄 

I'm privately working on a more generic version with c-shared lib that could potentially be used by other languages as well.

If you want to experiment with compiling nats-server to a c-shared lib you can get going pretty quickly by utilising this container based solution go cross compiler tool:  [xgo](https://github.com/crazy-max/xgo)

**Reply by isaaca26** [reply]
This is exactly what I am thinking. Thank you for sharing. Please keep me updated on your other project. Ideally it would be great if we could get the generic c-shared lib code incorporated in this nats server project. Maybe one of us could contribute the code?
